packages/hardhat-core/test/internal/hardhat-network/provider/node.ts

```diff
@@ -730,6 +737,144 @@ describe("HardhatNode", () => {
       });
     }
   });
+
+  it("should run calls in the right hardfork context", async function () {
+    // fork mainnet at the block when EIP-1559 activated (12965000), and try to
+    // run a call that specifies gas limits in EIP-1559 terms, but run that
+    // call one block earlier, and expect that call to fail because it should
+    // have specified gas limits in PRE-EIP-1559 terms.
```

Comment:
Addressed in https://github.com/nomiclabs/hardhat/pull/1928/commits/dbbd20df2f2932a654cee706782a957d780037da

Reply:
I still find this whole test very hard to understand. We should have at least one top-comment explaining the approach. Another way to make it clearer is to add error messages to the different assertions (something that, admittedly, we don't usually do, but it might be worth in this case to make the test more self-explanatory).

Reply:
Resolving this in light of https://github.com/nomiclabs/hardhat/pull/1928#discussion_r752680478